src/rust/src/wallet.rs

```diff
@@ -554,6 +570,42 @@ impl Wallet {
     pub fn note_commitment_tree_root(&self) -> MerkleHashOrchard {
         self.witness_tree.root()
     }
+
+    /// Fetches the information necessary to spend the note at the given `OutPoint`,
+    /// relative to the current root known to the wallet of the Orchard commitment tree.
+    ///
+    /// Returns `None` if the `OutPoint` is not known to the wallet, or the Orchard bundle
+    /// containing the note has not been passed to `Wallet::append_bundle_commitments`.
+    pub fn get_spend_info(&self, outpoint: OutPoint) -> Option<OrchardSpendInfo> {
+        // TODO: Take `confirmations` parameter and obtain the Merkle path to the root at
+        // that checkpoint, not the latest root.
+        self.wallet_received_notes
+            .get(&outpoint.txid)
+            .and_then(|tx_notes| tx_notes.decrypted_notes.get(&outpoint.action_idx))
+            .and_then(|dnote| {
+                self.key_store
+                    .ivk_for_address(&dnote.note.recipient())
```

Comment:
We're doing the retrieval of FVK for an address in a couple of places now, later we should add a convenience method for it; doesn't need to be done in this PR.